src/transformers/visitors/orchestrationInternalFunctionCallVisitor.ts

```diff
@@ -213,6 +213,9 @@ const internalCallVisitor = {
                 file.nodes.forEach(childNode => {
                  if(childNode.nodeType === 'FunctionDefinition') {
                    childNode.parameters.modifiedStateVariables = joinWithoutDupes(childNode.parameters.modifiedStateVariables, state.newParametersList);
+                   const modifiedNodes = childNode.parameters.modifiedStateVariables.map(node => node.name);
+                   if(state.decNode && !(modifiedNodes.includes(state.decNode.declarations[0].name))) 
+                   childNode.body.preStatements.splice(1, 0, state.decNode);
```

Comment:
Could you leave a comment here, explaining this change?

Reply:
done